erts/emulator/sys/common/erl_sys_common_misc.c

```diff
@@ -191,7 +191,7 @@ sys_double_to_chars_fast(double f, char *buffer, int buffer_size, int decimals,
     }
 
     /* Don't bother with optimizing too large numbers or too large precision */
-    if (af > MAX_FLOAT || decimals >= MAX_DECIMALS) {
+    if ((Uint64)af > MAX_FLOAT || decimals >= MAX_DECIMALS) {
```

Comment:
This invokes Undefined Behaviour whenever the float represents a value larger than (1 << 64) -1, which is in the 10^18 area -- there is no modulo arithmetic when casting a float to an unsigned integer.

Reply:
Yes, I noticed that and therefore closed the pull request.  The proper test would be:
af > MAX_FLOAT || decimals >= MAX_DECIMALS || isnan(af)
however, in light of your comment that Erlang doesn't support NaNs, this is not needed.